compiler/setup.py

```diff
@@ -463,7 +463,6 @@ def find_git_submodule_revision(submodule_path):
     },
     install_requires=[
         "numpy",
-        "PyYAML",
         "sympy",
```

Comment:
This `sympy` dep appears to be coming from torch-mlir's fx_importer. I left a comment where it was introduced: https://github.com/iree-org/iree/pull/17662#discussion_r1722496958. I think we could move it down into `extras_require`.

Reply:
Yeah, sympy is imported at https://github.com/llvm/torch-mlir/blob/7f886cc2703cb3deec6e95fb10b4aca00ef70da6/test/python/fx_importer/sympy_to_affine_expr_test.py#L9. Worth to try if we can move it into `extras_require`.